component/src/main/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestAuthenticator.java

```diff
@@ -0,0 +1,396 @@
+	 * Extra request sending to Pinterest userinfo end-point.
+	 *
+	 * @param url         user info endpoint.
+	 * @param accessToken access token.
+	 * @return user info
+	 * @throws ApplicationAuthenticatorException
+	 */
+	private String getUserInfo(String url, String accessToken) throws ApplicationAuthenticatorException {
+		if (log.isDebugEnabled()) {
+			log.debug("Sending the request for getting the user info");
+		}
+		StringBuilder stringBuilder = new StringBuilder();
+		BufferedReader bufferedReader = null;
+		HttpURLConnection httpConnection = null;
+		try {
+			URL obj = new URL(url + PinterestAuthenticatorConstants.QUESTION_MARK +
+			                  PinterestAuthenticatorConstants.ACCESS_TOKEN + PinterestAuthenticatorConstants.EQUAL +
+			                  accessToken);
+			URLConnection connection = obj.openConnection();
+			// Cast to a HttpURLConnection
+			if (connection instanceof HttpURLConnection) {
+				httpConnection = (HttpURLConnection) connection;
+				httpConnection.setRequestMethod(PinterestAuthenticatorConstants.HTTP_GET_METHOD);
+				bufferedReader = new BufferedReader(new InputStreamReader(httpConnection.getInputStream()));
+			} else {
+				throw new ApplicationAuthenticatorException("Exception while casting the HttpURLConnection");
+			}
+			String inputLine = bufferedReader.readLine();
+			while (inputLine != null) {
+				stringBuilder.append(inputLine).append("\n");
+				inputLine = bufferedReader.readLine();
+			}
+		} catch (MalformedURLException e) {
+			throw new ApplicationAuthenticatorException("Exception while generating the user info URL", e);
+		} catch (ProtocolException e) {
+			throw new ApplicationAuthenticatorException("Exception while setting the request method", e);
+		} catch (IOException e) {
+			throw new ApplicationAuthenticatorException("Exception while reading the response", e);
+		} finally {
+			IdentityIOStreamUtils.closeReader(bufferedReader);
+			if (httpConnection != null) {
+				httpConnection.disconnect();
+			}
+		}
+		if (log.isDebugEnabled()) {
+			log.debug("Receiving the response for the User info: " + stringBuilder.toString());
+		}
+		return stringBuilder.toString();
+	}
+
+	/**
+	 * Get the Pinterest specific claim dialect URI.
+	 *
+	 * @return Claim dialect URI.
+	 */
+	@Override
+	public String getClaimDialectURI() {
+		return PinterestAuthenticatorConstants.CLAIM_DIALECT_URI;
+	}
+
+	/**
+	 * @param token                   token
+	 * @param authenticatorProperties authenticatorProperties
+	 * @return claims
+	 */
+	private Map<ClaimMapping, String> buildClaims(OAuthClientResponse token,
+	                                              Map<String, String> authenticatorProperties)
+			throws ApplicationAuthenticatorException {
+		Map<ClaimMapping, String> claims = new HashMap<>();
+		String accessToken = token.getParam("access_token");
+		String url = getUserInfoEndpoint(token, authenticatorProperties);
+		String json;
+		try {
+			json = getUserInfo(url, accessToken);
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-pinterest/pull/1/commits/2d20b214a60d365dbfdad4f646691e271a536769